apps/test/unit/dance/danceReduxTest.ts

```diff
@@ -148,6 +148,38 @@ describe('danceRedux', function () {
         assert.deepEqual(store.getState().dance.selectedSong, selectedSong);
       });
 
+      it('initializes filtered songs correctly', async () => {
+        const selectSongOptions = {
+          isProjectLevel,
+          freePlay,
+          defaultSong,
+          selectedSong,
+          songSelection: ['a'],
+        };
+        await dispatch(
+          initSongs({
+            useRestrictedSongs,
+            selectSongOptions,
+            onAuthError,
+            onSongSelected,
+          })
+        );
+
+        assert.isTrue(queryParams.calledWithExactly('manifest'));
+        assert.isTrue(
+          getSongManifest.calledWithExactly(useRestrictedSongs, userManifest)
+        );
+        assert.isTrue(parseSongOptions.calledWithExactly([{id: 'a'}]));
+        assert.isTrue(
+          getSelectedSong.calledWithExactly([{id: 'a'}], selectSongOptions)
+        );
```

Comment:
These are the relevant 2 new test bits - just making sure that the filtered value (from the songSelection on line 157) actually happen.